system/api/create.go

```diff
@@ -242,3 +219,86 @@ func createContentHandler(res http.ResponseWriter, req *http.Request) {
 	}
 
 }
+
+func createContentHandlerJSON(res http.ResponseWriter, req *http.Request) (*url.Values, error) {
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	formData, err := jsonToURLValues(&body)
+	if err != nil {
+		return nil, err
+	}
+	return formData, nil
+}
+
+func createContentHandlerMultiPartForm(res http.ResponseWriter, req *http.Request) (*url.Values, error) {
+	err := req.ParseMultipartForm(1024 * 1024 * 4) // maxMemory 4MB
+	if err != nil {
+		return nil, err
+	}
+
+	ts := fmt.Sprintf("%d", int64(time.Nanosecond)*time.Now().UnixNano()/int64(time.Millisecond))
+	req.PostForm.Set("timestamp", ts)
+	req.PostForm.Set("updated", ts)
+
+	urlPaths, err := upload.StoreFiles(req)
+	if err != nil {
+		return nil, err
+	}
+
+	for name, urlPath := range urlPaths {
+		req.PostForm.Set(name, urlPath)
+	}
+
+	// check for any multi-value fields (ex. checkbox fields)
+	// and correctly format for db storage. Essentially, we need
+	// fieldX.0: value1, fieldX.1: value2 => fieldX: []string{value1, value2}
+	fieldOrderValue := make(map[string]map[string][]string)
+	for k, v := range req.PostForm {
```

Comment:
Since this is about checkboxes and multivalue form fields I omitted from JSON handling, I expect the user to supply the object well formated